README.md

```diff
@@ -1920,6 +1926,174 @@ map.addEntryListener(mapEventListener).then(function () {
+
+
+A map-wide event is fired as a result of a map-wide operation. For example, `ReplicatedMap.clear()`. You should use the `MapEventListener` type to listen to these events. A `MapEvent` object is passed to the listener function. 
+
+See the following example.
+
+```javascript
+var mapEventListener = {
+    mapCleared: function (mapEvent) {
+        console.log('Map Cleared:', mapEvent.numberOfAffectedEntries); // Map Cleared: 3
+    }
+};
+rmp.addEntryListener(mapEventListener).then(function () {
+    return rmp.put('1', 'Muhammet Ali');
+}).then(function () {
+    return rmp.put('2', 'Ahmet');
+}).then(function () {
+    return rmp.put('3', 'Furkan');
+}).then(function () {
+    return rmp.clear();
+});
+```
+
+#### 7.5.2.4. Listening for Queue Events
+
+In Queue, you can listen to events by using the functions in the `ItemListener` interface including `itemAdded` and `itemRemoved`. To listen to these events, you need to implement the `ItemEventListener` function that is an `ItemEvent` object is passed to.
+
+See the following example.
+
+```javascript
+var itemListener = {
+    itemAdded: function (itemEvent) {
+        console.log('Item Added:', itemEvent.item); // Item Added: Furkan
+    },
+    itemRemoved: function (itemEvent) {
+        console.log('Item Removed:', itemEvent.item); // Item Removed: Furkan
+    }
+};
+return queue.addItemListener(itemListener, true).then(function () {
+    return queue.add('Furkan');
+}).then(function () {
+    return queue.remove('Furkan');
+});
+```
+
+#### 7.5.2.5. Listening for Set Events
+
+In Set, you can listen to events by using the functions in the `ItemListener` interface including `itemAdded` and `itemRemoved`. To listen to these events, you need to implement the `ItemEventListener` function that is an `ItemEvent` object is passed to.
+
+See the following example.
+
+```javascript
+var itemListener = {
+    itemAdded: function (itemEvent) {
+        console.log('Item Added:', itemEvent.item); // Item Added: Furkan
+    },
+    itemRemoved: function (itemEvent) {
+        console.log('Item Removed:', itemEvent.item); // Item Removed: Furkan
+    }
+};
+return set.addItemListener(itemListener, true).then(function () {
+    return set.add('Furkan');
+}).then(function () {
+    return set.remove('Furkan');
+});
+```
+
+#### 7.5.2.6. Listening for List Events
+
+In List, you can listen to events by using the functions in the `ItemListener` interface including `itemAdded` and `itemRemoved`. To listen to these events, you need to implement the `ItemEventListener` function that is an `ItemEvent` object is passed to.
+
+See the following example.
+
+```javascript
+var itemListener = {
+    itemAdded: function (itemEvent) {
+        console.log('Item Added:', itemEvent.item); // Item Added: Furkan
+    },
+    itemRemoved: function (itemEvent) {
+        console.log('Item Removed:', itemEvent.item); // Item Removed: Furkan
+    }
+};
+return list.addItemListener(itemListener, true).then(function () {
```

Comment:
Should explain what the second argument does and why it matters. If they do not include value in the event, the message is shorter. Also the server does not have to serialize the value, the client does not have to deserialie the value. Therefore, whole conversation is faster.